src/responses.ts

```diff
@@ -97,3 +99,52 @@ export interface GetTransactionStatusRpcResponse {
   }
   receipts_outcome: any[]
 }
+
+export interface NearBlock {
+  hash: string
+  parentHash: string
+  number: number
+
+  timestamp: number
+  nonce: string
+  difficulty: number
+  _difficulty: BigNumber
+
+  gasLimit: BigNumber
+  gasUsed: BigNumber
+
+  miner: string
+  extraData: string
+
+  baseFeePerGas?: null | BigNumber
+}
+
+export interface NearBlockWithChunk
+  extends Pick<NearBlock, 'gasLimit' & 'gasUsed' & 'hash' & 'parentHash' & 'number' & 'timestamp'> {
```

Comment:
I'd say that `NearBlock` should extend `Pick<Block, 'hash' | 'parentHash' | 'number' | 'timestamp' | 'nonce' | 'miner' | 'gasLimit'|'gasUsed'>` and there add specific keys for what a NearBlock represents (eg: `shard_id`, `chunk_hash`, etc).

After that, `NearBlockWithChunk` can extend `NearBlock` and then add the interface for the `Chunk` itself. See https://github.com/near/near-api-js/blob/aa33ce8f20f95e95370d5fc0b7e61369e2f43e5b/src/providers/provider.ts#L159-L178.

We can actually use some of those interfaces if they work for us, since we are already importing the `near-api-js` package

Reply:
Got it!